extension/content.js

```diff
@@ -220,6 +220,46 @@ function showRecentlyPushedBranches() {
 	$('.repository-content').prepend(div);
 }
 
+// Add option for viewing diffs without whitespace changes
+function addDiffViewWithoutWhitespaceOption() {
+	if ($('.diff-options-content').length < 1 && $('.btn-group .selected[href*="diff="]').length < 1) {
+		return;
+	}
+
+	// Detect if PR or commit
+	let type = 'pr';
+
+	if ($('.btn-group .selected[href*="diff="]').length > 0) {
+		type = 'commit';
+	}
+
+	const optionElement = document.createElement('a');
+	const matchUrlArgs = new RegExp(/(\?|&)(w=1)(?=&|#|$)/i);
+	const urlArgs = window.location.search === '' ? '?w=1' : `${window.location.search}&w=1`;
+	const urlHash = window.location.hash || '';
+	let optionIsSet = false;
+	let optionHref = window.location.origin + window.location.pathname + urlArgs + urlHash;
+
+	if (matchUrlArgs.test(window.location.search) === true) {
+		optionIsSet = true;
+		optionHref = window.location.href.replace(/(\?|&)(w=1)(?=&|#|$)/, '');
+		$(optionElement).addClass('selected');
+	}
+
+	const optionElementContent = `${optionIsSet ? '<svg aria-hidden="true" class="octicon octicon-check" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path d="M12 5L4 13 0 9l1.5-1.5 2.5 2.5 6.5-6.5 1.5 1.5z"></path></svg>' : ''} No whitespace diff`;
+
+	$(optionElement).attr('href', optionHref);
+	$(optionElement).html(optionElementContent);
```

Comment:
To add to this, `$(optionElement)` should be assigned to a variable, so it can be re-used below in the `type === 'pr'` conditional.